src/raft.c

```diff
@@ -1304,13 +1310,46 @@ void callRaftPeriodic(RedisModuleCtx *ctx, void *arg)
         EntryCacheCompact(rr->logcache, rr->config.log_max_cache_size);
     }
 
-    /* Initiate snapshot if log size exceeds raft-log-file-max */
-    if (!rr->snapshot_in_progress && rr->config.log_max_file_size &&
-        raft_get_num_snapshottable_logs(rr->raft) > 0 &&
-        LogFileSize(rr->log) > rr->config.log_max_file_size) {
-        LOG_DEBUG("Raft log file size is %lu, initiating snapshot.",
-                  LogFileSize(rr->log));
-        initiateSnapshot(rr);
+    /* Initiate snapshot if log size exceeds raft-log-file-max
+     *
+     * Compaction happens in two steps:
+     * 1- We call LogCompactionBegin() and start writing entries to a new file.
+     * 2- Once we commit all the entries of the first file, we initiate the
+     *    snapshot.
+     */
+    if (!rr->snapshot_in_progress) {
+        bool start;
+        /* Step-1: Start compaction if we are over the file size limit or if
+         * there is a debug req. */
+        uint64_t limit = rr->config.log_max_file_size;
+        start = (rr->debug_req || (limit && LogFileSize(&rr->log) > limit));
```

Comment:
since you now call `LogCompactionBegin` from `cmdRaftDebug` there is no need to use `rr->debug_req` condition anymore

Reply:
ok, deleted it.